### PR TITLE
Add banker's deque implementation

### PR DESCRIPTION
**Pull request by Zipheir**
This merges my banker's deque implementation of SRFI 134, which was adapted from Shiro's two-list implementation. I've also replaced the old `ideque-tests.scm` file with tests from Shiro's implementation, since the old tests were completely out-of-date and broken.

### CODE REVIEW

File: srfi-134.html [hunk 1]
```diff
@@ -353,16 +353,16 @@ <h2 id="Conversion">Conversion</h2>
...
+<p>The sample implementation is in the <code>srfi</code> subdirectory of the repository for
+this SRFI.  It uses a two-stream banker's deque structure (see
+<em>Purely Functional Data Structures</em> by
+Chris Okasaki, p. 108) which provides amortized O(1) running time
+for basic deque operations.
+An implementation based on finger trees may be added later.</p>
 
 <h1>Copyright</h1>
 Copyright (C) John Cowan, Kevin Wortman (2015).  All Rights Reserved.
+Copyright (C) Wolfgang Corcoran-Mathe (2022).  All Rights Reserved.
```
Comment (arthurgleckler) [inline srfi-134.html]: This copyright notice is for the SRFI document, not the code.  You've just changed one paragraph, so it's probably not right to add your name here.  What do you think?

Comment (Zipheir) [reply srfi-134.html]: Oops, sorry about that. I'll revert that. I also found an arithmetic error in ideque-take, so standby for a new PR.

**Comment by Zipheir** [comment]
OK, no need to open a new PR. Does this look good?

**Comment by Zipheir** [comment]
One more thing to fix. I forgot to read the exact semantics of `ideque=` before simplifying it.

**Comment by Zipheir** [comment]
Fixed.

**Pull request merged**